vision/google/cloud/vision/annotations.py

```diff
@@ -89,8 +89,11 @@ def from_api_repr(cls, response):
         annotations = {}
         for feature_type, annotation in response.items():
             curr_feature = annotations.setdefault(_KEY_MAP[feature_type], [])
-            curr_feature.extend(
-                _entity_from_response_type(feature_type, annotation))
+            entity = _entity_from_response_type(feature_type, annotation)
+            if isinstance(entity, list):
+                curr_feature.extend(entity)
+            else:
+                annotations[_KEY_MAP[feature_type]] = entity
```

Comment:
OK, re-reading this, it seems it could benefit from a helper function and then just be:

```python
annotations = {
    _KEY_MAP[feature_type]: _toms_helper(feature_type, annotation),
    for feature_type, annotation in six.iteritems(response)
}
return cls(**annotations)
```

I **WAS** worried about key collision, but after examining the structure it's clear there wouldn't be one (provided the values in `_KEY_MAP` are unique).

I was also thrown off by the fact that you grab

```python
curr_feature = annotations.setdefault(_KEY_MAP[feature_type], [])
```

independent of the value of `isinstance(entity, list)` (you only need it in the `True` branch).